ai-gateway/concepts/api-keys.mdx

```diff
@@ -0,0 +1,131 @@
+---
+title: Provider API Keys
+description: Understanding provider API key management in the AI Gateway
+---
+
+Provider API keys authenticate requests to AI providers like OpenAI, Anthropic, and Google. The AI Gateway offers flexible key management—from simple passthrough to centralized management with automatic failover.
+
+## Key precedence
+
+The gateway determines which provider API key to use based on these rules:
+
+### 1. Gateway keys override SDK keys
+
+If you configure provider API keys in the gateway, those keys are used instead of any key sent by your SDK:
+
+```yaml
+providers:
+  - id: "openai"
+    api_keys:
+      - value: ${secrets.get('openai', 'gateway-key')}

```

Comment:
suggestion: I know we want to push people into this workflow, but do you think we should add a note here along the lines of "Securely load your api key from an ngrok secret instead of keeping it plain text" or something to explain the "complexity" of this macro and sell people on the usefulness of the feature?

Reply:
I do this in other locations, we could just port that over